R/ashm_multi_rainbow_plot.R

```diff
@@ -0,0 +1,146 @@
+                      "chr22:23217074-23250428","chr2:89073691-89320640", "chr8:128774985-128876311","chr18:60982124-60990180")
+  regions_bed = data.frame(regions = regions, names = names)
+  regions_bed = dplyr::filter(regions_bed, names %in% regions_to_display)
+  regions = pull(regions_bed, regions)
+  names = pull(regions_bed, names)
+  if(missing(maf_data)){
+    region_mafs = lapply(regions, function(x){get_ssm_by_region(region = x, streamlined = TRUE,seq_type = seq_type)})
+  }else{
+    region_mafs = lapply(regions, function(x){get_ssm_by_region(region = x, streamlined = TRUE, maf_data = maf_data)})
+  }
+  tibbled_data = tibble(region_mafs, region_name = names)
+  unnested_df = tibbled_data %>%
+    unnest_longer(region_mafs)
+
+  unlisted_df = mutate(unnested_df, start = region_mafs$Start_Position, sample_id = region_mafs$Tumor_Sample_Barcode) %>%
+    dplyr::select(start,sample_id, region_name)
+
+  meta_arranged = meta_arranged %>% mutate_if(is.factor, as.character)
+  meta_arranged = meta_arranged %>% mutate(classification = factor(!!sym(classification_column)))
+
+
+  muts_anno = left_join(unlisted_df, meta_arranged)
+  muts_first = dplyr::select(muts_anno, start, region_name) %>%
+    group_by(region_name) %>%
+    arrange(start) %>%
+    dplyr::filter(row_number() == 1)
+
+  eg = expand_grid(start = pull(muts_first, start), sample_id = pull(meta_arranged, sample_id))
+  eg = left_join(eg, muts_first)
+
+  #concatenate expanded frame of points with original mutation data
+  real_and_fake = bind_rows(unlisted_df, eg)
+  muts_anno = left_join(real_and_fake, meta_arranged)
+
+  muts_anno$sample_id = factor(muts_anno$sample_id, levels = meta_arranged$sample_id)
+
+  if(!missing(regions_to_display)){
+    muts_anno = dplyr::filter(muts_anno, region_name %in% regions_to_display)
+  }
+  #make the plot
+  p = muts_anno %>%
+        ggplot() +
+        geom_point(aes(x = start, y = sample_id, colour = classification), alpha = 0.4, size = 0.6) +
+        labs(title = "", subtitle = "", x = "", y = "Sample") +
+        theme_Morons() +
+        theme(axis.text.y = element_blank(), axis.ticks.y = element_blank(), plot.margin = ggplot2::margin(1,1,1,1, "cm"), title = element_blank(), plot.subtitle = element_blank(), axis.title.x = element_blank()) +
+        facet_wrap(~region_name, scales = "free_x") +
+        guides(color = guide_legend(reverse = TRUE,
+                                    override.aes = list(size = 3),
+                                    title={{ classification_column }}))
+
+  if(! missing(custom_colours)){
+    # ensure only relevant color keys are present
+    custom_colours = custom_colours[intersect(names(custom_colours), pull(meta_arranged[,classification_column]))]
+    p = p +
+        scale_colour_manual(values = custom_colours)
+  }
+
+  print(p)
+
+}
```

Comment:
Would these function-level R scripts also need a blank line inserted at the end of the script, or is this fine?

Reply:
We would need a blank line at the end, for sure. This is EOF (end of file) and signals where the file ends. Otherwise, git has no way of finding out whether the file is complete or not

Reply:
Is there a way that we can reproduce error due to the lack of EOF? 
I could install the package and the functions are working fine. I can change the code to add a blank line at the end and generate the files again though.

Reply:
There won't be an error message - exactly because git doesn't know whether this is end of file or not. But it will show the warning sign like this:
![image](https://github.com/morinlab/GAMBLR.viz/assets/47579823/789aaf5f-0729-4b0b-aebc-9749e50a750c)
This file seem to have the EOF line so we are good - just something to look for at the future

Reply:
It also looks like all the files in this PR have the EOF line so no need to change anything. :+1: